zenml/core/backends/processing/processing_dataflow_backend.py

```diff
@@ -71,32 +71,48 @@ def __init__(
         self.disk_size_gb = disk_size_gb
         self.autoscaling_algorithm = autoscaling_algorithm
         self.image = image
-        super().__init__(**kwargs)
+        self.staging_location = staging_location
+        self.temp_location = temp_location
+
+        # staging and temp can be the same if one is specified
+        if staging_location is not None and temp_location is None:
+            self.temp_location = self.staging_location
+
+        super().__init__(
+            project=project,
+            staging_location=staging_location,
+            temp_location=temp_location,
+            region=region,
+            job_name=job_name,
+            image=image,
+            machine_type=machine_type,
+            num_workers=num_workers,
+            max_num_workers=max_num_workers,
+            disk_size_gb=disk_size_gb,
+            autoscaling_algorithm=autoscaling_algorithm,
+        )
 
     def get_beam_args(self,
                       pipeline_name: Text = None,
                       pipeline_root: Text = None) -> \
             Optional[List[Text]]:
-        temp_location = os.path.join(pipeline_root, 'tmp', pipeline_name)
-        stage_location = os.path.join(pipeline_root, 'staging', pipeline_name)
+        if self.temp_location is None:
+            self.temp_location = os.path.join(
+                pipeline_root, 'tmp', pipeline_name)
+        if self.staging_location is None:
+            self.staging_location = os.path.join(
+                pipeline_root, 'staging', pipeline_name)
 
         return [
-            '--runner=DataflowRunner',
+            '--runner=dataflow',
             '--project=' + self.project,
-            '--temp_location=' + temp_location,
-            '--staging_location=' + stage_location,
+            '--temp_location=' + self.temp_location,
+            '--staging_location=' + self.staging_location,
             '--region=' + self.region,
             # '--job_name=' + self.job_name,
             '--num_workers=' + str(self.num_workers),
             '--max_num_workers=' + str(self.max_num_workers),
 
-            # Specifying dependencies
```

Comment:
It is such a big relief to see this part removed. Perhaps we can clean the utils accordingly and remove all the unused additional functions such as prepare_sdist.

Reply:
yes!